test/Desalt.Core.Tests/Translation/TranslatorTests.Declarations.cs

```diff
@@ -333,6 +333,43 @@ class A {
 ");
         }
 
+        [Test]
+        [Ignore("Not yet implemented")]
+        public async Task Translate_should_put_methods_of_GlobalMethods_classes_in_global_namespace_in_declarations()
+        {
+            await AssertTranslation(
+                @"
+[GlobalMethods]
+static class A
+{
+    public static void Method1() { }
+
+    [PreserveName]
+    static void Method2() {}
+
+    [PreserveCase]
+    static void Method3() {}
+
+    [ScriptName(""Renamed"")]
+    static void Method4() { }
+
+    [InlineCode(""FooModule.foo({a})"")]
+    static void Method5(int a) { }
+
+    [ScriptSkip]
+    static void Method6(int a) { }
+}",
+@"
+global.method1 = function(): void { };
```

Comment:
I wasn't sure how you wanted to handle things getting put in the global namespace...

Reply:
The generated code matches what Saltarelle currently does. However, it ties us to mscorlib.js, which defines the 'global' variable. It's at the very top of mscorlib:

```
if (typeof (global) === 'undefined') {
  if (typeof (window) !== 'undefined')
    global = window;
  else if (typeof (self) !== 'undefined')
    global = self;
}
```

What I was thinking is to make this even simpler and create plain functions. Since the generated TypeScript lives in a top-level file, I was thinking something like this:

```
export function method1() { }

export function method2() { }

export function Method3() { }

export function Renamed() { }
```

You'll have to determine whether or not the `export` should be added by seeing if the method is publicly visible (the method itself is visible and the containing hierarchy of classes). There's already a helper method `ExportIfNeeded` that determines whether something should be exported or not. It's in the `Translator` class.